posthog/warehouse/api/data_warehouse.py

```diff
@@ -85,13 +85,71 @@ def total_rows_stats(self, request: Request, *args: Any, **kwargs: Any) -> Respo
         return Response(
             status=status.HTTP_200_OK,
             data={
-                "billingAvailable": billing_available,
-                "billingInterval": billing_interval,
-                "billingPeriodEnd": billing_period_end,
-                "billingPeriodStart": billing_period_start,
-                "materializedRowsInBillingPeriod": materialized_rows,
-                "totalRows": rows_synced,
-                "trackedBillingRows": billing_tracked_rows,
-                "pendingBillingRows": pending_billing_rows,
+                "billing_available": billing_available,
+                "billing_interval": billing_interval,
+                "billing_period_end": billing_period_end,
+                "billing_period_start": billing_period_start,
+                "materialized_rows_in_billing_period": materialized_rows,
+                "total_rows": rows_synced,
+                "tracked_billing_rows": billing_tracked_rows,
+                "pending_billing_rows": pending_billing_rows,
             },
         )
+
+    @action(methods=["GET"], detail=False)
+    def recent_activity(self, request: Request, **kwargs) -> Response:
+        class DataWarehouseActivityRow(TypedDict):
+            id: str
+            type: str
+            name: Optional[str]
+            status: str
+            rows: int
+            created_at: str
+            finished_at: Optional[str]
+            latest_error: Optional[str]
+            workflow_run_id: Optional[str]
+
+        limit = min(int(request.GET.get("limit", 50)), 250)
```

Comment:
limits could probably be less? Not sure how much activity needs to be default pulled